packages/connection-storage/src/connection-storage.ts

```diff
@@ -1,520 +1,102 @@
-      // remove favorites and the just saved connection (so we do not delete it)
-      .filter((x) => !x.favorite && x.id !== savedConnection.id)
-      .sort((a, b) => {
-        const aTime = a.lastUsed?.getTime() ?? 0;
-        const bTime = b.lastUsed?.getTime() ?? 0;
-        return bTime - aTime;
-      });
-
-    if (recentConnections.length >= this.maxAllowedRecentConnections) {
-      await this.delete({
-        id: recentConnections[recentConnections.length - 1].id,
-      });
-    }
-  }
-
-  static async deserializeConnections({
-    content,
-    options = {},
-    signal,
-  }: {
+  deserializeConnections?(args: {
     content: string;
     options: ImportConnectionOptions;
     signal?: AbortSignal;
-  }): Promise<ConnectionInfo[]> {
-    throwIfAborted(signal);
+  }): Promise<ConnectionInfo[]>;
 
-    const { passphrase, trackingProps, filterConnectionIds } = options;
-    const connections = await deserializeConnections(content, {
-      passphrase,
-      trackingProps,
-    });
-    return filterConnectionIds
-      ? connections.filter((x) => filterConnectionIds.includes(x.id))
-      : connections;
-  }
+  exportConnections?(args?: {
+    options?: ExportConnectionOptions;
+    signal?: AbortSignal;
+  }): Promise<string>;
 
-  static async importConnections({
-    content,
-    options = {},
-    signal,
-  }: {
+  importConnections?(args: {
     content: string;
     options?: ImportConnectionOptions;
     signal?: AbortSignal;
-  }): Promise<void> {
-    throwIfAborted(signal);
-
-    const connections = await this.deserializeConnections({
-      content,
-      options,
-      signal,
-    });
-
-    log.info(
-      mongoLogId(1_001_000_149),
-      'Connection Import',
-      'Starting connection import',
-      {
-        count: connections.length,
-      }
-    );
-
-    await Promise.all(
-      connections.map((connectionInfo) => this.save({ connectionInfo, signal }))
-    );
-
-    log.info(
-      mongoLogId(1_001_000_150),
-      'Connection Import',
-      'Connection import complete',
-      {
-        count: connections.length,
-      }
-    );
-  }
+  }): Promise<void>;
+}
 
-  static async exportConnections({
-    options = {},
-    signal,
-  }: {
-    options?: ExportConnectionOptions;
-    signal?: AbortSignal;
-  } = {}): Promise<string> {
-    throwIfAborted(signal);
+export type ConnectionStorageIPCInterface = Required<
+  Omit<ConnectionStorage, 'on' | 'off' | 'emit'>
+>;
 
-    const { filterConnectionIds, ...restOfOptions } = options;
-    const connections = await this.loadAll({ signal });
-    const exportConnections = filterConnectionIds
-      ? connections.filter((x) => filterConnectionIds.includes(x.id))
-      : connections.filter((x) => x.favorite?.name);
+export type ConnectionStorageIPCMain = Pick<HadronIpcMain, 'createHandle'>;
 
-    return serializeConnections(exportConnections, restOfOptions);
-  }
-}
+export type ConnectionStorageIPCRenderer = Pick<
```

Comment:
This seems a bit too platform specific for this file. What's the reason to have it here?

Reply:
I was just having all the types in one place but yea I get the point, will move it to a separate file or even maybe main storage itself.